packages/http/src/decorators.ts

```diff
@@ -122,6 +125,38 @@ export function isHeader(program: Program, entity: Type) {
   return program.stateMap(HttpStateKeys.header).has(entity);
 }
 
+export const $cookie: CookieDecorator = (
+  context: DecoratorContext,
+  entity: ModelProperty,
+  cookieNameOrOptions?: string | CookieOptions,
+) => {
+  const paramName =
+    typeof cookieNameOrOptions === "string"
+      ? cookieNameOrOptions
+      : (cookieNameOrOptions?.name ??
+        entity.name.replace(/([a-z])([A-Z])/g, "$1_$2").toLowerCase());
+  const userOptions = typeof cookieNameOrOptions === "object" ? cookieNameOrOptions : {};
+  const options: CookieParameterOptions = {
+    type: "cookie",
+    name: paramName,
+    explode: userOptions.explode ?? true,
+    format: "form",
+  };
+  context.program.stateMap(HttpStateKeys.cookie).set(entity, options);
+};
+
+export function getCookieParamOptions(program: Program, entity: Type): QueryParameterOptions {
+  return program.stateMap(HttpStateKeys.cookie).get(entity);
+}
+
+export function getCookieParamName(program: Program, entity: Type): string {
```

Comment:
i would maybe omit this function as it was added there for the others before we had any additional options